src/bitdrift_public/protobuf/filter/v1/filter.proto

```diff
@@ -62,10 +64,62 @@ message Filter {
         MATCHING_ONLY = 2;
       }
 
+      message SetFieldValue {
+        message ExistingField {
+          // The name of an existing field.
+          string name = 1 [(validate.rules).string = {min_len: 1}];
+        }
+
+        oneof value {
+          // The constant string value.
+          string string_value = 1;
+          // The value of an existing field.
+          ExistingField existing_field = 2;
+        }
+      }
+
       string name = 1 [(validate.rules).string = {min_len: 1}];
-      string value = 2 [(validate.rules).string = {min_len: 1}];
+      // The value to set the field to.
+      SetFieldValue value = 2;
+      // The type of the field to set.
       FieldType field_type = 3 [(validate.rules).enum = {defined_only: true not_in: 0}];
+      // Whether a transform is allowed to override an existing field. If no and the field already exists,
+      // the transform is a no-op.
+      bool allow_overriding = 4;
     }
+
+    // Removes a specified field(s).
+    message RemoveFields {
```

Comment:
since you already have transforms be a list maybe just have this be RemoveField? unless you wanted to evolve this into something else eventaully?